### ISSUE TITLE
Remove unnecessary null pointer checks

### ISSUE THREAD

**Issue by elfring** [michaelrsweet/mxml#276]
[Extra null pointer checks are not needed in functions](https://stackoverflow.com/questions/18775608/free-a-null-pointer-anyway-or-check-first "Free a null pointer anyway or check first?") like the following.
- [mxml_free](https://github.com/michaelrsweet/mxml/blob/371bb0c0a0af3cd5ce883aa713f2dd522a0428c2/mxml-node.c#L775 "mxml_free function")
- [mxmlIndexDelete](https://github.com/michaelrsweet/mxml/blob/371bb0c0a0af3cd5ce883aa713f2dd522a0428c2/mxml-index.c#L49 "mxmlIndexDelete function")

**Comment by michaelrsweet** [comment]
Agreed, target platforms for Mini-XML are not old C89 (or older) compilers anymore.

**Comment by elfring** [comment]
I am curious then if another software developer (besides me) can get into the mood to apply a corresponding update suggestion which could be generated by [the software “Coccinelle”](https://coccinelle.gitlabpages.inria.fr/website/ "Source code transformation with Coccinelle") (also with the help of a variant from the following script).

```cocci
@Remove_unnecessary_pointer_checks@
expression x;
@@
-if (\(x != 0 \| x != NULL\))
    free(x);
```

**Comment by michaelrsweet** [comment]
@elfring When it comes to detecting potential coding errors or inefficient patterns, I have no problem running a search or scan. But I have a hard time “blindly” trusting a global search-and-replace of this scope since in my experience it is easy to introduce coding/logic errors that way.

**Comment by elfring** [comment]
:thinking: How do you think about to try a possible update generation out for the mentioned transformation pattern with the help of the semantic patch language (Coccinelle software)?
